src/Components/Search/NewSearch/NewSearchBarInput.tsx

```diff
@@ -1,70 +1,184 @@
+        value: option.displayLabel!,
+        subtitle:
+          getLabel({
+            displayType: option.displayType ?? "",
+            typename: option.__typename,
+          }) ?? "",
+        imageUrl: option.imageUrl!,
+        showArtworksButton: !!option.statuses?.artworks,
+        showAuctionResultsButton: !!option.statuses?.auctionLots,
+        href: option.href!,
         typename: option.__typename,
-      }),
-      imageUrl: option.imageUrl!,
-      showArtworksButton: !!option.statuses?.artworks,
-      showAuctionResultsButton: !!option.statuses?.auctionLots,
-      href: option.href!,
+        item_number: index,
+        item_type: option.displayType!,
+      }
     }
-  })
+  )
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setValue(event.target.value)
+  useUpdateEffect(() => {
+    tracking.trackEvent({
+      action_type:
+        options.length > 0
+          ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
+          : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+      query: value,
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchIncrement])
+
+  const refetch = (value: string, entity?: SearchEntity) => {
+    const entities = entity ? [entity] : []
+    const performanceStart = performance && performance.now()
 
-    //TODO: Add throttle
     relay.refetch(
       {
         hasTerm: true,
-        term: event.target.value,
+        term: value,
+        entities: entities,
       },
       null,
       error => {
         if (error) {
           logger.error(error)
           return
         }
+
+        if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
+          reportPerformanceMeasurement(performanceStart)
+        }
+
+        setSearchIncrement(prevCounter => prevCounter + 1)
       }
     )
   }
+  const throttledSearchRequest = useMemo(
+    () => throttle(refetch, SEARCH_THROTTLE_DELAY, { leading: true }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value)
+    throttledSearchRequest(event.target.value)
+  }
+
+  const handlePillClick = (pill: PillType) => {
+    setSelectedPill(pill)
+    refetch(value, ELASTIC_PILL_KEY_TO_SEARCH_ENTITY?.[pill.key])
+  }
+
+  const handleRedirect = () => {
+    setValue("")
+  }
+
+  const handleSubmit = () => {
+    if (value) {
+      window.location.href = `/search?term=${encodeURIComponent(value)}`
+    }
+  }
+
+  const handleSelect = (option: any) => {
+    tracking.trackEvent({
+      action_type: DeprecatedSchema.ActionType.SelectedItemFromSearch,
+      destination_path:
+        option.typename === "Artist"
+          ? `${option.href}/works-for-sale`
+          : option.href,
+      item_number: option.item_number,
+      item_type: option.item_type,
+      query: value,
+    })
+
+    setValue("")
+    window.location.href = option.href
```

Comment:
Yep, for the record, anytime we use window.location.whatever, we lose all routing history (along with our routing session caches), and its quite a hit to performance. 

Always lean on the router for any kind of navigation case except for cases where things are so complicated that you literally have to reboot the computer (so to speak) -- aka, hard jumps between pages. (The one example I can think of where this is necessary is logging in and out.)

Reply:
Not sure if we can get rid of `AutocompleteInputOption` easily, we will have to make changes in palette, and I'm not sure if I like this change. For now I went with using `router.push()` instead of `window.location`, but happy to re-iterate later

Reply:
But I think we will have to do it... Clicking on quick nav links doesn't work, I think because of what you mentioned @dimatretyak